### PR TITLE
Azure AD Identity and Access (Independent Publisher)

### PR DESCRIPTION
**Pull request by Laskewitz**
---
When submitting a connector, please check the following conditions for your PR.

- [x] I attest that the connector works and I verified by deploying and testing all the operations.
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a logo.

### CODE REVIEW

**Comment by Laskewitz** [comment]
@sriyen-msft when will this one be checked? 🙂

File: independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,193 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure AD Invitation Manager",
+    "description": "A connector for the Azure AD Invitation Manager.",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json:5]: @Laskewitz Can you please provide a more detailed description ?

Docs - [Description of connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector)

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json (hunk 1 above)
Comment (Laskewitz) [reply independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json]: @sriyen-msft fixed!

**Review by Laskewitz — COMMENTED** [review]
Description added

File: independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json (hunk 1 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json]: @Laskewitz Connector's description should describe concisely the main purpose and value offered by your connector. The give description is not enough to understand what is it created for. Please provide detailed explanation about `Azure AD Invitation Manager`.

File: independent-publisher-connectors/AADInvitationManager/apiProperties.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        },
+        "uiDefinition": {
+          "constraints": {
+            "required": "false",
+            "hidden": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#ffffff",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AADInvitationManager/apiProperties.json:42]: @Laskewitz Did you read the checklist mentioned in the description of your PR ?

image

You can provide `"#da3b01"` which is default iconBrandColor of Individual Publisher connector.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json (hunk 1 above)
Comment (Laskewitz) [reply independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json]: @sriyen-msft in a later commit I fixed the description for the connector. See: https://github.com/microsoft/PowerPlatformConnectors/pull/924/commits/2398a7be54af473c206258aa61dd02c66122fa3b

File: independent-publisher-connectors/AADInvitationManager/apiProperties.json (hunk 2 above)
Comment (Laskewitz) [reply independent-publisher-connectors/AADInvitationManager/apiProperties.json]: Fixed in https://github.com/microsoft/PowerPlatformConnectors/pull/924/commits/a70220abaa9f57a919431035fd626b6d97d91e25

**Comment by Laskewitz** [comment]
Can I get an update on this? This is taking very - very long this way.

**Comment by Laskewitz** [comment]
@natalie-pienkowska changed the connector name as promised. Good to go now.

File: independent-publisher-connectors/AADInvitationManager/readme.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+## Azure AD Identity and Access
+
+The Graph API is a powerful and very extensive REST API. Using this API, you can create and manage objects (e.g. users, groups etc.) in Azure Active Directory, create Microsoft Teams teams, and a lot more.
+
+This connector exposes the Identity and Access resources of the Graph API in the Microsoft Power Platform.
+
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AADInvitationManager/readme.md:6]: @Laskewitz The publisher section is missing, You can find a template for readme.md [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/readme.md). Can you please add it and other sections if you have valid data.

File: independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                          "emailAddress": {
+                            "$ref": "#/definitions/emailAddress"
+                          }
+                        }
+                      },
+                      "description": "Additional recipients the invitation message should be sent to. Currently only 1 additional recipient is supported."
+                    },
+                    "customizedMessageBody": {
+                      "type": "string",
+                      "description": "customizedMessageBody"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json:78]: @Laskewitz All the description values are expected to be free from grammatical and spell errors. Please provide valid description values for all the properties.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json (hunk 4 above)
Comment (Laskewitz) [reply independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json:78]: Fixed

File: independent-publisher-connectors/AADInvitationManager/readme.md (hunk 3 above)
Comment (Laskewitz) [reply independent-publisher-connectors/AADInvitationManager/readme.md:6]: Fixed

**Comment by Laskewitz** [comment]
@sriyen-msft Committed the changes. So all good to go now.

File: independent-publisher-connectors/AADInvitationManager/readme.md (hunk 3 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/AADInvitationManager/readme.md:6]: @Laskewitz The publisher section is missing which is a required one. Can you please let me know if you are not able to access the template I have shared in my previous message?

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by Laskewitz** [comment]
@sriyen-msft It does help when I push the changes as well ;-) Now it should be OK...

**Comment by sriyen-msft** [comment]
@Laskewitz We will move forward with preparing your connector for the next deployment schedule. Once your connector is onboards the next deployment schedule, it will deploy your connector in our production environments which would typically take between 7-10 business days. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps. Thanks

**Comment by sriyen-msft** [comment]
[[certify-connector]]

**Comment by sriyen-msft** [comment]
@Laskewitz  We are prepping your connector production deployment; we need to set up an OAuth app and add its client id and secret values to our Production KeyVault. We are unable to set up an OAuth app using our test account as it doesn’t have the required permissions to Azure AD.  

Since you are an internal partner, can you please set up the OAuth app and share the client id and secret to us in a secure channel, so that we can upload them to our Prod KV to proceed with production deployment.

@natalie-pienkowska and @mamurshe  - FYI

**Comment by sriyen-msft** [comment]
@Laskewitz Deployment freeze notice: Due to Thanksgiving week in the United States, we have temporarily halted the deployments of all the connectors till Tuesday, 30th November 2021.

We will begin the deployments on Wednesday, 1st December 2021. If your connector is queued up for Wednesday (1st December 2021) deployment schedule, the production deployment of your connector in all regions will be expected to be completed by Thursday, 9th December 2021. Thanks

**Comment by natalie-pienkowska** [comment]
Our team is currently working on setting up OAuth for this connector. We will provide updates and comments along the way.

**Comment by Laskewitz** [comment]
Any update on this?

**Comment by natalie-pienkowska** [comment]
Hi @Laskewitz this connector is on the deployment train. It should be fully deployed by February 7th :)

**Comment by natalie-pienkowska** [comment]
Unfortunately, we had some issues with the deployment. @mamurshe is on top of this and will do everything he can to deploy these connectors.

**Comment by sriyen-msft** [comment]
@Laskewitz Apologies for the delay in response, We are actively working on preparing and deploying your connector to publish environments. As you know that we are deploying the connector as a Standard connector, For that, we need to go thru a few guidelines and approvals from the concerned teams (ex. Power BI API team) which is a bit lengthy process. Please allow us some more time, we will keep you posted with the updates and deployment plan soon. Thank you so much for your patience.

**Comment by Amjed-Ayoub** [comment]
Dear partner,

We apologize for the delay you might be experiencing while we are working on a high volume of connector requests. Rest assured that we are doing everything we can to address this volume. We are taking measures like improving our processes and hiring additional team members to guarantee resolving your connector requests in a timely manner. 

Please let us know if you have any questions in the meantime.

File: independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  },
+  "parameters": {},
+  "responses": {},
+  "securityDefinitions": {
+    "undefined": {
+      "type": "oauth2",
+      "flow": "accessCode",
+      "authorizationUrl": "https://login.windows.net/common/oauth2/authorize",
+      "tokenUrl": "https://login.windows.net/common/oauth2/authorize",
+      "scopes": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AADInvitationManager/apiDefinition.swagger.json:184]: Hi @Laskewitz I hope you are doing well.
Apologies for the delay in the process.
Can you please add the scopes that you are targeting the Graph API for, and be more specific?
Thank you very much.

File: independent-publisher-connectors/AADInvitationManager/apiProperties.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{
+  "properties": {
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "aad",
+          "clientId": "",
+          "scopes": [],
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/AADInvitationManager/apiProperties.json:9]: Please add the specific scopes. Thank you very much

**Comment by Amjed-Ayoub** [comment]
Hello @Laskewitz,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please accept our apologies for the delay in the process.
Please let us know if you have any questions.
Thank you very much for working with us.

**Review by sriyen-msft — APPROVED** [review]

**Comment by sriyen-msft** [comment]
@Laskewitz Apologies for the delay in response. 

Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Pull request merged**